labelindex/named_port_index.go

```diff
@@ -255,24 +270,43 @@ func (idx *SelectorAndNamedPortIndex) OnUpdate(update api.Update) (_ bool) {
 	return
 }
 
-func convertIPs(a, b []net.IP) []ip.Addr {
-	combined := make([]ip.Addr, 0, len(a)+len(b))
-	for _, addr := range a {
-		combined = append(combined, ip.FromNetIP(addr.IP))
+// extractCIDRsFromHostEndpoint converts the expected IPs of the host endpoint into /32 and /128
+// CIDRs.
+func extractCIDRsFromHostEndpoint(endpoint *model.HostEndpoint) []ip.CIDR {
+	v4Addrs := endpoint.ExpectedIPv4Addrs
+	v6Addrs := endpoint.ExpectedIPv6Addrs
+	combined := make([]ip.CIDR, 0, len(v4Addrs)+len(v6Addrs))
+	for _, addr := range v4Addrs {
+		combined = append(combined, ip.FromNetIP(addr.IP).AsCIDR())
 	}
-	for _, addr := range b {
-		combined = append(combined, ip.FromNetIP(addr.IP))
+	for _, addr := range v6Addrs {
+		combined = append(combined, ip.FromNetIP(addr.IP).AsCIDR())
 	}
 	return combined
 }
 
-func convertNets(a, b []net.IPNet) []ip.Addr {
-	combined := make([]ip.Addr, 0, len(a)+len(b))
-	for _, addr := range a {
-		combined = append(combined, ip.FromNetIP(addr.IP))
+// extractCIDRsFromWorkloadEndpoint converts the IPv[46]Nets fields of the WorkloadEndpoint into
+// /32 and /128 CIDRs.  It ignores any prefix length (but our validation ensures those nets are
+// /32s or /128s in any case).
```

Comment:
Why not propagate the prefix length here?  ISTM that would be in the right direction for adding e.g. IPv6 prefix delegation.

Reply:
I did consider it but I was worried about making that change without giving it enough thought.  I _think_ the validation prevents non-/32s from popping out here so we could change this part but I'm pretty sure that downstream requires /32s right now.

Reply:
OK, that makes sense.  If we did implement non-/32s, there'd be no chance of missing that we had to relax this code again.